addons/mail/static/src/components/activity/tests/activity_tests.js

```diff
@@ -1276,6 +1276,52 @@ QUnit.test('data-oe-id & data-oe-model link redirection on click', async functio
     );
 });
 
+QUnit.test('chatter is updated when activity type is changed', async function (assert) {
+    assert.expect(2);
+
+    const activityId = 513;
+    this.data['res.partner'].records.push({
+        activity_ids: [activityId],
+        id: 100,
+    });
+    this.data['mail.activity'].records.push({
+        activity_category: 'upload_file',
+        activity_type_id: 28,
+        can_write: true,
+        id: activityId,
+        res_id: 100,
+        res_model: 'res.partner',
+    });
+    const { createChatterContainerComponent } = await this.start();
+    await createChatterContainerComponent({
+        threadId: 100,
+        threadModel: 'res.partner',
+    });
+
+    await this.env.services.rpc({
+        model: 'mail.activity',
+        method: 'write',
+        args: [[activityId], {
+            activity_category: 'default',
+            activity_type_id: 1,
+        }],
+    });
+    await afterNextRender(async () => {
+        const activity = this.messaging.models['Activity'].findFromIdentifyingData({ id: activityId });
+        await activity.fetchAndUpdate();
+    });
```

Comment:
Would briefly explain why we need manual fetch and update, just because this looks off compared to many other tests.